src/java/org/apache/cassandra/hints/HintsWriter.java

```diff
@@ -248,7 +249,10 @@ void append(Hint hint) throws IOException
                 updateChecksumInt(crc, hintSize);
                 out.writeInt((int) crc.getValue());
 
+                long startPosition = out.position();
                 Hint.serializer.serialize(hint, out, descriptor.messagingVersion());
+                long actualSize = out.position() - startPosition;
+                checkState(actualSize == hintSize);
```

Comment:
can we have a useful error msg?  If this triggers we don't have enough info on what happened